src/snowflake/snowpark/_internal/data_source/drivers/pyodbc_driver.py

```diff
@@ -82,6 +84,30 @@ def to_snow_type(self, schema: List[Any]) -> StructType:
             fields.append(StructField(name, data_type, null_ok))
         return StructType(fields)
 
+    def udtf_class_builder(self, fetch_size: int = 1000) -> type:
+        create_connection = self.create_connection
+
+        def binary_converter(value):
+            return value.hex() if value is not None else None
+
+        class UDTFIngestion:
+            def process(self, query: str):
+                import pyodbc
+
+                conn = create_connection()
+                conn.add_output_converter(pyodbc.SQL_BINARY, binary_converter)
+                conn.add_output_converter(pyodbc.SQL_VARBINARY, binary_converter)
+                conn.add_output_converter(pyodbc.SQL_LONGVARBINARY, binary_converter)
```

Comment:
same question as oracledb, what if user provides output converter

Reply:
added, there is one thing to notice is that the function to get current converter only exist in recent versions, we need to specify that in our dependency or let user know it: https://github.com/mkleehammer/pyodbc/wiki/Connection#get_output_converter